ga4gh/avrotools.py

```diff
@@ -152,9 +223,194 @@ def _getInvalidFields(cls, expectedSchema, datum):
             for field in expectedSchema.fields:
                 key = field.name
                 value = datum.get(key)
-                result = cls._getInvalidFields(field.type, value)
-                if result != cls.sinkValue:
+                result = self._getInvalidFields(field.type, value)
+                if result != self.sinkValue:
                     dic[key] = result
             if len(dic):
                 return dic
-        return cls.sinkValue
+        return self.sinkValue
+
+
+class RandomSchemaTool(object):
```

Comment:
I'm fine with changing this name, but unfortunately `SchemaTool`'s name needs to be pretty generic because its function is pretty generic.  It just takes a schema and then does a variety of things with it, depending on the methods called... validation, generating an instance, etc.  A less descriptive name is the consequence of moving towards a "one class that has all of the schema-related methods in it" approach, which is what this refactor is all about (as opposed to the "too java-y" classmethods approach)!